k8s/controllers/actions/catalogeval_controller.go

```diff
@@ -40,6 +43,10 @@ func (r *CatalogEvalReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	evalCR := &federationv1.CatalogEvalExpression{}
 
+	resourceK8SId := evalCR.GetNamespace() + "/" + evalCR.GetName()
+	operationName := fmt.Sprintf("%s/%s", constants.CatalogEvalOperationNamePrefix, constants.ActivityOperation_Delete)
```

Comment:
You need to use DeletionTimestamp.IsZero() to determine the delete and write action.